### PR TITLE
Add text to audio task

### PR DESCRIPTION
**Pull request by Vaibhavs10**

### CODE REVIEW

**Comment by osanseviero** [comment]
You will also need to update in a couple of other places https://github.com/search?q=repo%3Ahuggingface%2Fhub-docs+TEXT-to-speech+language%3ATypeScript&type=code&l=TypeScript

**Comment by Vaibhavs10** [comment]
I think I added it to all the relevant ones. Do you mind giving it a look again please @osanseviero ?

**Review by osanseviero — COMMENTED** [review]
Awesome, things are looking good! You will also need to

* Add an icon (similar to https://github.com/huggingface/hub-docs/blob/main/js/src/lib/components/Icons/IconTextToSpeech.svelte ) and specify it in https://github.com/huggingface/hub-docs/blob/main/js/src/lib/components/PipelineIcon/PipelineIcon.svelte#L27 (I suggest to check https://carbondesignsystem.com/ for icons)
* You might also want to enable the widget, which involves modifying https://github.com/huggingface/hub-docs/blob/main/js/src/lib/components/InferenceWidget/InferenceWidget.svelte (most likely you can re-use `TextToSpeechWidget`)

You can preview how this would look like in GitHub Codespaces as specified in https://github.com/huggingface/moon-landing#codespace

**Comment by Vaibhavs10** [comment]
Noice. Going to to test this now!

**Comment by Vaibhavs10** [comment]
Weird! I keep getting permission denied on codespace for anything I do. Is this the same for you too @osanseviero ?

**Comment by Vaibhavs10** [comment]
Hey @mishig25 

After your suggestions, I tried running `npm ci` and `npm run prettier` - no changes
then I tried `npm run lint` also resulted in no changes.

Send help! - welp!

**Comment by Vaibhavs10** [comment]
Hey @mishig25 - I just copied InferenceTextToSpeech widget over to text-to-audio.
So there are no new changes. I'll follow up with a separate PR to change the actual look.

Let me know if this is okay for you?

**Comment by osanseviero** [comment]
I think we should reuse `TextToSpeechWidget` rather than reimplementing it, same way we reuse `TabularDataWidget`. API input/output is exactly the same

**Comment by mishig25** [comment]
Btw, is the task already supported in api-inference? Is there a model or curl script to test on ?

**Comment by Vaibhavs10** [comment]
Hey! @mishig25 - Sorry for the delayed response, `api-inference` needed an update to make sure that the model can work.

Since the pipeline_tag `text-to-audio` is not yet supported the models cannot be directly curled, however, changing the pipeline_tag to `text-to-speech` does the trick and works - you can test it here: https://huggingface.co/reach-vb/musicgen-small?text=lo-fi+music

Do note that this is embarrassingly slow - will work on fixing that separately.

@osanseviero - I was thinking of making a follow-up PR to update the Icon for TTM, that's why I created those files. I think it'd be good to distinguish the two. However, if you feel strongly about this then I can revert the change. lmk.

I'll treat this as priority today, would be nice to get this merged soon.

**Comment by Vaibhavs10** [comment]
Note: that `text-to-speech` is an alias to `text-to-audio` pipeline so the behaviour would be exactly the same for both.

**Review by osanseviero — COMMENTED** [review]
Ok from my side to follow-up with the icon, but as mentioned before, we should reuse the widget, no need to implement a new one

**Comment by Vaibhavs10** [comment]
@osanseviero - Removed all the custom `TextToAudio` code. I am waiting for the checks to run.

Is there anything else left for this to be good to merge? @mishig25 @osanseviero

**Comment by Vaibhavs10** [comment]
Both the checks have passed! 🤗

**Review by osanseviero — APPROVED** [review]
LGTM to merge if the TTA Inference API is working already

**Comment by Vaibhavs10** [comment]
Good to merge, then?

For testing the inference API, feel free to head over to this URL: https://huggingface.co/reach-vb/musicgen-small

File: js/src/lib/components/PipelineIcon/PipelineIcon.svelte
Comment (mishig25) [inline js/src/lib/components/PipelineIcon/PipelineIcon.svelte:63]: if it is exact same icon as `IconTextToSpeech`, why not just use `IconTextToSpeech` ?

**Comment by osanseviero** [comment]
It seems stuck in loading or internal server errors though

File: js/src/lib/components/PipelineIcon/PipelineIcon.svelte
Comment (Vaibhavs10) [reply js/src/lib/components/PipelineIcon/PipelineIcon.svelte]: Good call: I fixed it in the latest commit.

**Comment by Vaibhavs10** [comment]
@osanseviero -> The model is pretty much unusable on CPU (it takes an insane amount of time to generate).

I tested it from `api-inference` main - It returns a file as expected!
image

The plan is to enable these for GPU run-time once the pipeline is merged.

**Pull request merged**
